lib/_http_outgoing.js

```diff
@@ -283,17 +285,26 @@ OutgoingMessage.prototype._send = function _send(data, encoding, callback) {
 OutgoingMessage.prototype._writeRaw = _writeRaw;
 function _writeRaw(data, encoding, callback) {
   const conn = this.socket;
-  if (conn && conn.destroyed) {
-    // The socket was destroyed. If we're still trying to write to it,
-    // then we haven't gotten the 'close' event yet.
-    return false;
-  }
 
   if (typeof encoding === 'function') {
     callback = encoding;
     encoding = null;
   }
 
+  if (conn && conn.destroyed) {
+    const err = new ERR_STREAM_DESTROYED('write');
+    if (callback) {
+      callback(err);
+    }
+    if (this.listenerCount('error') > 0 && this[kErrored] === undefined) {
+      this[kErrored] = true;
+      this.emit('error', err);
```

Comment:
I believe this should be emitted in next tick? Otherwise e.g. `writeContinue` will call the callback synchronously, which is unexpected.

Reply:
Maybe same thing with callback?

Reply:
I'm actually removing calling the callback here. It's not needed to fix the actual issue, and it can be done in a later step.

Reply:
We cannot remove the callback or something else is going to fail. Essentially if we want to emit error here, we need to call the callback _synchronously_ first :/.

Reply:
Something else, as in a test? Add a comment for future semver-major?